spec/models/candidate_interface/rejection_reasons_history_spec.rb

```diff
@@ -35,12 +35,105 @@
       feedback = described_class.all_previous_applications(current_application_form, :becoming_a_teacher)
 
       expect(feedback).to match_array [
-        described_class::HistoryItem.new(choice3.provider.name, :becoming_a_teacher, 'Amazing'),
-        described_class::HistoryItem.new(choice2.provider.name, :becoming_a_teacher, 'Good'),
-        described_class::HistoryItem.new(choice1.provider.name, :becoming_a_teacher, 'Bad'),
+        described_class::HistoryItem.new(choice3.provider.name, :becoming_a_teacher, 'Amazing', 'reasons_for_rejection'),
+        described_class::HistoryItem.new(choice2.provider.name, :becoming_a_teacher, 'Good', 'reasons_for_rejection'),
+        described_class::HistoryItem.new(choice1.provider.name, :becoming_a_teacher, 'Bad', 'reasons_for_rejection'),
       ]
     end
 
+    context 'when redesigned rejection reasons' do
+      let(:previous_application_form) { create(:application_form) }
+      let!(:application_choice1) { create(:application_choice, :with_redesigned_rejection_reasons, application_form: previous_application_form, structured_rejection_reasons: rejection_reasons) }
+      let(:current_application_form) { apply_again!(previous_application_form) }
+      let!(:application_choice2) { create(:application_choice, :with_redesigned_rejection_reasons, application_form: current_application_form) }
+
+      context 'when no reasons for section selected' do
+        let(:rejection_reasons) { { selected_reasons: [{ id: 'course_full', label: 'Course full' }] } }
+
+        it 'returns nothing' do
+          feedback = described_class.all_previous_applications(current_application_form, :becoming_a_teacher)
+
+          expect(feedback).to be_empty
+        end
+      end
+
+      context 'when reasons selected for the `becoming_a_teacher` section' do
+        let(:rejection_reasons) do
+          {
+            selected_reasons: [
+              {
+                id: 'personal_statement',
+                label: 'Personal statement',
+                selected_reasons: [
+                  {
+                    id: 'quality_of_writing',
+                    label: 'Quality of writing',
+                    details: {
+                      id: 'quality_of_writing_details',
+                      text: 'Quality Bad',
+                    },
+                  },
+                  {
+                    id: 'personal_statement_other',
+                    label: 'Other',
+                    details: {
+                      id: 'personal_statement_other_details',
+                      text: 'Personal statement Bad',
+                    },
+                  },
+                ],
+              },
+            ],
+          }
+        end
+
+        it 'returns the related rejections in history items' do
+          feedback = described_class.all_previous_applications(current_application_form, :becoming_a_teacher)
+
+          history_item = feedback.first
+          expect(feedback.count).to eq(1)
```

Comment:
readability option: `expect(feedback).to have(1).item`

Reply:
Unfortunately the `have` matchers are part of the `rspec-collection_matchers` gem (which is not included in this repo) and have been removed from RSpec 3